posthog/temporal/batch_exports/pre_export_stage.py

```diff
@@ -0,0 +1,601 @@
+                )
+            except Exception as e:
+                await logger.aexception("Unexpected error occurred while deleting existing objects from S3", exc_info=e)
+                raise
+
+            query_id = uuid.uuid4()
+            await logger.ainfo(f"Executing query with ID = {query_id}")
+            try:
+                await client.execute_query(query, query_parameters=query_parameters, query_id=str(query_id))
+            except Exception as e:
+                await logger.aexception("Unexpected error occurred while writing record batches to S3", exc_info=e)
+                raise
+
+
+class ProducerFromInternalS3Stage:
+    """
+    This is an alernative implementation of the `spmc.Producer` class that reads data from the internal S3 staging area.
+    """
+
+    def __init__(self):
+        self.logger = get_internal_logger()
+        self._task: asyncio.Task | None = None
+
+    @property
+    def task(self) -> asyncio.Task:
+        if self._task is None:
+            raise ValueError("Producer task is not initialized, have you called `ProducerFromInternalS3Stage.start()`?")
+        return self._task
+
+    async def start(
+        self,
+        queue: RecordBatchQueue,
+        batch_export_id: str,
+        data_interval_start: str | None,
+        data_interval_end,
+        max_record_batch_size_bytes: int = 0,
+        min_records_per_batch: int = 100,
+    ) -> asyncio.Task:
+        self._task = asyncio.create_task(
+            self.produce_batch_export_record_batches_from_range(
+                queue=queue,
+                batch_export_id=batch_export_id,
+                data_interval_start=data_interval_start,
+                data_interval_end=data_interval_end,
+                max_record_batch_size_bytes=max_record_batch_size_bytes,
+                min_records_per_batch=min_records_per_batch,
+            ),
+            name="record_batch_producer",
+        )
+        return self._task
+
+    async def produce_batch_export_record_batches_from_range(
+        self,
+        queue: RecordBatchQueue,
+        batch_export_id: str,
+        data_interval_start: str | None,
+        data_interval_end: str,
+        max_record_batch_size_bytes: int = 0,
+        min_records_per_batch: int = 100,
+    ):
+        folder = _get_s3_staging_folder(
+            batch_export_id=batch_export_id,
+            data_interval_start=data_interval_start,
+            data_interval_end=data_interval_end,
+        )
+
+        session = aioboto3.Session()
+        async with session.client(
+            "s3",
+            # TODO - should we use our own set of env vars for this?
+            # TODO - check these are available in production workers
+            aws_access_key_id=settings.OBJECT_STORAGE_ACCESS_KEY_ID,
+            aws_secret_access_key=settings.OBJECT_STORAGE_SECRET_ACCESS_KEY,
+            endpoint_url=settings.OBJECT_STORAGE_ENDPOINT,
+            region_name=settings.OBJECT_STORAGE_REGION,
```

Comment:
I am not familiar with these env variables, so I think we should look up what they are used for, just in case they don't get changed from under us.

Reply:
May be easier to just use our own.

Reply:
will update this